### PR TITLE
fix(activity): correct counter on revert and load-more (#1019, #1021)

### LINKED ISSUE #1019: fix: activity history 'showing' counter regresses to first-page size after revert
## Summary

After a user loads more activity history entries (via "Load more"), reverting a change resets the "showing X of Y" counter to the first-page size.

## Root cause

The activity feed's "Load more" button does not use \`hx-push-url\`, so \`HX-Current-URL\` always has \`offset=0\` even after multiple pages have been loaded. The revert fragment derives the counter from \`activeFilter.Offset\` parsed from \`HX-Current-URL\`, so after a revert the counter is always \`limit\` (first-page size), not the actual number of visible rows.

The artist-history tab has the same pattern: \`List(..., limit, 0)\` is called with hard-coded offset=0, so \`showing = len(changes)\` regresses to the first-page count.

## Fix options

- Track visible count client-side (e.g. count \`#activity-entry\` DOM nodes) and pass via \`hx-vals\` to the revert endpoint
- Or accept the counter as approximate after load-more (cosmetic issue only)

## References

Flagged in PR #1016 review (Copilot comments 3068776610, 3068776608; CodeRabbit review 4094282567).

### LINKED ISSUE #1021: fix: undo/revert OOB counter shows stale page-1 count after load-more
## Problem

After a user loads more activity rows (via load-more), triggering undo on any visible row sends a revert request. The handler reconstructs the `showing` counter using `activeFilter.Offset + limit` derived from the current page URL. The initial page URL still has `offset=0`, so the OOB counter replacement shows the first-page count (e.g. "20 of 47") instead of the actual visible count (e.g. "35 of 47").

## Steps to reproduce

1. Open the activity feed with more than `pageSize` entries.
2. Click "Load more" to append additional rows.
3. Undo any entry visible after the initial page.
4. Observe the "Showing N of M" counter reverts to first-page state.

## Proposed fix

Pass the current visible count with the revert request (e.g. via a hidden `hx-vals` param on each row), and use that value to reconstruct the counter in `handleRevertHistory`. Alternatively, skip the OOB counter update entirely when the visible count is unknown.

## Context

Identified during code review of PR #1016. Deferred because the fix requires a new request parameter and coordinated load-more fragment changes.

### PR DESCRIPTION
**Pull request by sydlexius**
## Summary
- Extract `resolveShowingCount` so both `handleRevertHistory` render branches honor the hx-vals showing hint identically (was inconsistent: only the artist-tab branch logged malformed hints).
- Lock in the load-more counter regression with a unit test of the helper plus handler-level integration tests covering hint-preferred, fallback, and rejection paths for both the activity feed and the artist history tab.

## Test plan
- [x] `/activity` -> Load more -> Revert a recent entry -> "Showing X of Y" reflects DOM row count, not page-1 size (UAT confirmed)
- [x] `/activity` -> Revert top entry without load-more -> Counter unchanged (UAT confirmed)
- [x] Artist detail page -> History tab counter path covered by `TestHandleRevertHistory_ArtistTabShowingCounter_LoadMoreHint` (unit; not UAT-driven)
- [x] `bash scripts/pre-push-gate.sh` clean

Closes #1019
Closes #1021

## Summary by CodeRabbit

* **Refactor**
  * Unified history “Showing X of Y” counter so activity feed and artist history show consistent counts; client hints are validated and invalid or out-of-range hints fall back to a safe server-derived value (with debug logging).

* **Tests**
  * Added unit and integration tests for counter selection, boundary cases, malformed/negative hints, and view-specific fragment rendering.

### CODE REVIEW

File: internal/api/handlers_history.go [hunk 1]
```diff
@@ -122,6 +123,46 @@ func buildGlobalFilter(req *http.Request, limit int) artist.GlobalHistoryFilter
...
+	}
+	if n <= 0 || n > total {
+		// A non-positive count is meaningless (the user clicked undo, so at
+		// least one row was visible). A count exceeding total would render
+		// "Showing N of M" with N > M, which is nonsense; reject and use
+		// the fallback. Both cases also short-circuit a tampered hint.
+		return fallback
+	}
+	return n
+}
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/api/handlers_history.go:174]: Fixed in 8f236cd9. Clamped `fallback` to [0, total] at `resolveShowingCount` entry — hint validation already rejected out-of-range values; this extends the same guarantee to the fallback path. Note: CodeRabbit's '✅ Addressed in commits 6647d49 to 839d455' footer was incorrect — the rebase didn't add the clamping, only this commit does.

File: internal/api/handlers_history_test.go [hunk 2]
```diff
@@ -701,3 +703,349 @@ func TestHandleActivityContent_RendersRow(t *testing.T) {
...
+// tab) call to honor the hx-vals "showing" hint while degrading gracefully
+// when the hint is missing, malformed, or out of range.
+func TestResolveShowingCount(t *testing.T) {
+	logger := slog.New(slog.NewTextHandler(os.Stderr, &slog.HandlerOptions{Level: slog.LevelError}))
+
+	mkReq := func(showing string) *http.Request {
+		body := url.Values{}
+		if showing != "" {
+			body.Set("showing", showing)
+		}
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/api/handlers_history_test.go]: Fixed in 8f236cd9. `mkReq` now takes an `includeShowing bool` so empty-hint and missing-key cases are distinct, plus two new cases that exercise the clamp branches in `resolveShowingCount`. Patch coverage is now 97.5% on this PR.

**Comment by sydlexius** [comment]
@coderabbitai resolve
